src/Core/Math/Rectangle.js

```diff
@@ -3,14 +3,61 @@
  * Class: Rectangle
  */
 
-function Rectangle() {
```

Comment:
Instead of RasterDataTileProvider, we can also select io_driver according to input info when layer is added? so that providers contain only an instance of io_driver?  regardless of data format?